src/components/List.js

```diff
@@ -121,7 +130,11 @@ const List = ({ token }) => {
                 //disabled={checkPurchasedDate(doc.lastPurchased)} // Commented out for PR purposes only
               />
               {doc.itemName}
-              <p>Time until next purchase {doc.durationOfTime}</p>
+              {doc.numberOfPurchases > 0 ? (
+                <p>Time until next purchase {doc.durationOfTime}</p>
```

Comment:
@veeteeran @LINDGRENBA the first time I check the checkbox, the "time until next purchase" text shows up but it doesn't display a value. Should it be displaying something?
![purchase](https://user-images.githubusercontent.com/2433616/106983005-0d025780-671a-11eb-98d6-44a39de07424.PNG)

Once I click it a second time, then it starts to show the value

Reply:
Hi @chrisdopuch . One thing I wasn't sure was clear is that the value is only there to make it easier for reviewers to see that it's working and won't be part of the final UI/UX. Lauren suggested it in one of the previous comments. I still think this was a good thing to point out because it may be an indicator of the code not functioning as we expected, but it won't be part of the final UI.... unless the team likes it and wants to keep it in, but I think we deal with organizing the items by this value in one of next week's issues and there's a different approach 👍 😄

Reply:
I may be totally misunderstanding things (as I often do... ;-P), but it looks like right now we're saving to the DB a human readable form of the time to next purchase at the exact time that the user checked the checkbox. That means that that value will never change, even as time marches on. I believe that we may only need to save the estimated next purchase date and the number of times that the item has been purchased in the database. 

We should be able to derive the time until the next purchase from the distance between now and the estimated next purchase date and we can just calculate that on the fly via a function in our component. My general rule for determining whether or not something should be saved to a DB (with every rule there are always exceptions...) is that if I can derive it from other pre-existing values, then there's no need to save it. 

However, as you mentioned, this is just for demo purposes, so it's not a biggie either way. :-)

Reply:
Correcting myself slightly as @laurenmbeatty rightly pointed out we are supposed to be recording the estimated days until next purchase, not the next purchase date. I would take that to mean the number of days, which, if added onto the last purchase date, will give us our next purchase date. Thus, if we want to display how much time a user has at any given moment until they should be purchasing again, we would not want to directly display the value in the DB, as that won't change, but rather we would display the time between now and that next purchase date.

Reply:
@MikeBLambert I agree with what @laurenmbeatty and you are saying about the values saved in the database. In the database we want to save `numberOfPurchases` and `daysUntilNextPurchase`.

What is time? 🤣